### DISCUSSION TITLE
how to make a two lines x-axis (year and date) using ggh4x?

### DISCUSSION THREAD

**Discussion by yippiekahee** [teunbrand/ggh4x#133] [Q&A] [answered]
I try to use ggh4x to create straightforward understandable plots using a double x-axis: first, a 'normal' dates, shown in %m-format. Directly beneath I would like to have the year. So I do not have to show the same year on the x-axis repetitively.

The best I can accomplish is:

if(!require(tidyverse)) install.packages("tidyverse")
library(tidyverse)
if(!require(ggh4x)) install.packages("ggh4x")
library(ggh4x)

start_date <- as.Date("2022-01-01")
end_date <- as.Date("2023-12-31")
random_dates <- sample(seq(start_date, end_date, by="days"), 10, replace=TRUE)
random_values <- runif(10, min = 0, max = 20)
df <- data.frame(date = random_dates, value = random_values)

df %<>% 
  mutate(date = as.Date(date)) %>%
  mutate(year = format(date, "%Y"))
df

gg <- ggplot(df) +
  geom_point(aes(x = interaction(date, year), y = value, group = 1)) +
  scale_x_discrete(NULL, guide = "axis_nested")  
gg

This generates more or less what I want, but it shows non-equidistant dates as equidistant. Probably due to scale_x_discrete. Is there a way to solve this?
![plot](https://github.com/teunbrand/ggh4x/assets/32000999/2bd9aa00-28c0-47c0-aa1a-2b8d88a9f7f3)

**Comment by teunbrand** [comment, marked as answer]
You can use the `date_breaks` argument in `scale_x_date()` to separate year from day/month with a special delimiter, and then let `guide_axis_nested()` know about this delimiter.

``` r
library(dplyr)
library(ggplot2)
library(ggh4x)

start_date <- as.Date("2022-01-01")
end_date <- as.Date("2023-12-31")
random_dates <- sample(seq(start_date, end_date, by="days"), 10, replace=TRUE)
random_values <- runif(10, min = 0, max = 20)
df <- data.frame(date = random_dates, value = random_values)

df %<>%
  mutate(date = as.Date(date)) %>%
  mutate(year = format(date, "%Y"))

ggplot(df) +
  geom_point(aes(x = date, y = value, group = 1)) +
  scale_x_date(
    date_labels = "%d-%m+%Y",
    guide = guide_axis_nested(delim = "+")
  )
```

![](https://i.imgur.com/Ve29qmk.png)

<sup>Created on 2023-11-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Reply by yippiekahee** [reply]
Thank you, quite an elegant solution. This 'special delimiter' is a nice option.
